### PR TITLE
Rename opentelemetry-extension-incubator to opentelemetry-api-incubator, merge opentelemetry-api-events

### LINKED ISSUE #6121: Merge opentelemetry-api-events into opentelemetry-extension-incubator
> Writing this made me realize that its silly that we incubate APIs in both `opentelemetry-api-events` and `opentelemetry-extension-incubator`. I propose we merge events into `opentelemetry-extension-incubator`, and do it in lockstep with #6001 to minimize churn.

_Originally posted by @jack-berg in https://github.com/open-telemetry/opentelemetry-java/pull/6120#discussion_r1443197913_

### PR DESCRIPTION
**Pull request by jack-berg**
Resolves #6121. Summary of changes:

- BREAKING: Rename `io.opentelemetry:opentelemetry-extension-incubator` to `io.opentelemetry:opentelemetry-api-incubator`
- BREAKING: Rename packages `io.opentelemetry.extension.incubator.*` to `io.opentelemetry.api.incubator`
- BREAKING: Stop publishing `io.opentelemetry:opentelemetry-api-event`. Merge contents into `io.opentelemetry:opentelemetry-api-incubator`.
- BREAKING: Rename package `io.opentelemetry.api.events ` to `io.opentelemetry.api.incubator.events`

Part of the [goal of bundling several breaking changes](https://github.com/open-telemetry/opentelemetry-java/pull/6253#issuecomment-1971346317) into a single release to reduce churn.

cc @breedx-splk

### CODE REVIEW

File: api/incubator/build.gradle.kts
Comment (jack-berg) [inline api/incubator/build.gradle.kts:10]: The package changes as well.

**Comment by jack-berg** [comment]
I considered making an equivalent change to rename `opentelemetery-sdk-extension-incubator` to `opentelemetry-sdk-incubator`, but realized that `opentelemetry-sdk-extension-incubator` is different in that it actually does incubate SDK extensions, not just SDK code. For example, it incubates [view file config](https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk-extensions/incubator#view-file-configuration) and [file config](https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk-extensions/incubator#file-configuration), both of which are autoconfigure extensions.

**Comment by jack-berg** [comment]
FYI, if we can't get this PR and a followup to be able to define event payloads in the log record body in time for the next release, I think we need to revert #6253.

**Review by breedx-splk — APPROVED** [review]
Yeah, I support this. Sorry about the delay in getting it rubber stamped....kubecon timing is a little tricky. I hope we can still get the remaining event stuff in! 🤞🏻

**Comment by jack-berg** [comment]
@open-telemetry/java-approvers I'll merge this tomorrow if there are no additional comments. Need to keep moving on this to bundle all the changes w/ churn in time for the next release.

**Pull request merged**
